Adds environments component (#180)
carsdotcom/skelebot: docs/environments.md

Environment configurations can override any property in the skelebot yaml, even jobs, but keep in mind that any attribute in the environemtn config will completely override the attribute from the default config. This means if you want to change a child attribute, everything under that parent will need to be included in the config as well.

> All of the available environments for a given Skelebot project can be listed with the `envs` command. The `[default]` value that appears first int he list indicates the global skelebot.yaml file, and is active by default when no `-e` parameter is passed.

Reviewer: ```suggestion
All of the available environments for a given Skelebot project can be listed with the `envs` command. The `[default]` value that appears first in the list indicates the global skelebot.yaml file, and is active by default when no `-e` parameter is passed.
```
